parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetOutputFormat.java

```diff
@@ -341,22 +362,32 @@ private static int getMaxPaddingSize(Configuration conf) {
         throws IOException, InterruptedException {
     final WriteSupport<T> writeSupport = getWriteSupport(conf);
 
+    ParquetProperties props = ParquetProperties.builder()
+        .withPageSize(getPageSize(conf))
+        .withDictionaryPageSize(getDictionaryPageSize(conf))
+        .withDictionaryEncoding(getEnableDictionary(conf))
+        .withWriterVersion(getWriterVersion(conf))
+        .estimateRowCountForPageSizeCheck(getEstimatePageSizeCheck(conf))
+        .withMinRowCountForPageSizeCheck(getMinRowCountForPageSizeCheck(conf))
+        .withMaxRowCountForPageSizeCheck(getMaxRowCountForPageSizeCheck(conf))
+        .build();
+
     long blockSize = getLongBlockSize(conf);
-    if (INFO) LOG.info("Parquet block size to " + blockSize);
-    int pageSize = getPageSize(conf);
-    if (INFO) LOG.info("Parquet page size to " + pageSize);
-    int dictionaryPageSize = getDictionaryPageSize(conf);
-    if (INFO) LOG.info("Parquet dictionary page size to " + dictionaryPageSize);
-    boolean enableDictionary = getEnableDictionary(conf);
-    if (INFO) LOG.info("Dictionary is " + (enableDictionary ? "on" : "off"));
+    int maxPaddingSize = getMaxPaddingSize(conf);
     boolean validating = getValidation(conf);
+
+    if (INFO) LOG.info("Parquet block size to " + blockSize);
+    if (INFO) LOG.info("Parquet page size to " + props.getPageSizeThreshold());
+    if (INFO) LOG.info("Parquet dictionary page size to " + props.getDictionaryPageSizeThreshold());
+    if (INFO) LOG.info("Dictionary is " + (props.isEnableDictionary() ? "on" : "off"));
     if (INFO) LOG.info("Validation is " + (validating ? "on" : "off"));
-    WriterVersion writerVersion = getWriterVersion(conf);
-    if (INFO) LOG.info("Writer version is: " + writerVersion);
-    int maxPaddingSize = getMaxPaddingSize(conf);
+    if (INFO) LOG.info("Writer version is: " + props.getWriterVersion());
     if (INFO) LOG.info("Maximum row group padding size is " + maxPaddingSize + " bytes");
+    if (INFO) LOG.info("Page size checking is: " + (props.estimateNextSizeCheck() ? "estimated" : "constant"));
+    if (INFO) LOG.info("Min row count for page size check is: " + props.getMinRowCountForPageSizeCheck());
+    if (INFO) LOG.info("Min row count for page size check is: " + props.getMaxRowCountForPageSizeCheck());
```

Comment:
I thought about it, but the problem is that this is only printed once and not all of the settings are in parquet properties so it doesn't entirely clean it up. I think we should leave as-is for now and refactor later if we end up needing to reuse it.